MilitaryManager.Core/Services/UnitService.cs

```diff
@@ -58,7 +73,11 @@ public async Task<UnitDTO> UpdateUnitAsync(UnitDTO query)
 
         public async Task<UnitDTO> DeleteUnitAsync(int id)
         {
-            var unit = new Unit() { Id = id };
+            var unit = await _unitRepository.GetFirstBySpecAsync(new Units.UnitById(id));
```

Comment:
Is it necessary to include all the related properties to the unit for deleting it?

Reply:
It only deletes the unit itself, no related properties are deleted

Reply:
You can get just a unit without including related props.